### DISCUSSION TITLE
Get EffectiveName of DynamicBlockReference

### DISCUSSION THREAD

**Discussion by kwollaston** [CEXT-Dan/PyRx#75] [Q&A] [answered]
I have been struggling trying to get the EffectiveName for a BlockReference, specifically of a Dynamic Block.
I was able to get the attributes, but the only name I can get back is "*Uxx"

```for id in selection.objectIds():  
  bref = Db.BlockReference(id)
  brec = Db.BlockTableRecord(bref.blockTableRecord())
  print(brec.name()) # on the brec it works, on the bref it does not
  # I want the name shown in the GUI and not the anonymous name of dynamic block
```

As an example, I'm able to get the effective name using win32com by doing this:
```for entity in selection:
  print(entity.EffectiveName)
```

Let me know if you need more information. This is a great project and I appreciate all the work you and others have put in!

**Comment by CEXT-Dan** [comment, marked as answer]
bref.getBlockName() should return the effective name if it's dynamic. 

if that doesn't work then maybe try
```python
dynref = Db.DynBlockReference(id)
dynid = dynref.dynamicBlockTableRecord()
```

**Reply by kwollaston** [reply]
Perfect, bref.getBlockName() is the ticket, works for dynamic blocks and static blocks.
